src/components/panel/panel.js

```diff
@@ -17,6 +18,118 @@ angular
+ * will be stored within the provider itself. When the configuraiton is needed,
+ * the user can request for the configuration object. Configurations are set at
+ * the provider level and retrieved from the service level.
+ *
+ * @usage
+ * <hljs lang="js">
+ * (function(angular, undefined) {
+ *   'use strict';
+ *
+ *   angular
+ *       .module('demoApp', ['ngMaterial'])
+ *       .config(DemoConfig)
+ *       .controller('DemoCtrl', DemoCtrl)
+ *       .controller('DemoMenuCtrl', DemoMenuCtrl);
+ *
+ *   function DemoConfig($mdPanelConfigProvider) {
+ *     var config = {
+ *       name: 'demoConfig',
+ *       options: {
+ *         attachTo: angular.element(document.body),
+ *         controller: DemoMenuCtrl,
+ *         controllerAs: 'ctrl',
+ *         template: '' +
+ *             '<div class="menu-panel" md-whiteframe="4">' +
+ *             '  <div class="menu-content">' +
+ *             '    <div class="menu-item" ng-repeat="item in ctrl.items">' +
+ *             '      <button class="md-button">' +
+ *             '        <span>{{item}}</span>' +
+ *             '      </button>' +
+ *             '    </div>' +
+ *             '    <md-divider></md-divider>' +
+ *             '    <div class="menu-item">' +
+ *             '      <button class="md-button" ng-click="ctrl.closeMenu()">' +
+ *             '        <span>Close Menu</span>' +
+ *             '      </button>' +
+ *             '    </div>' +
+ *             '  </div>' +
+ *             '</div>',
+ *         panelClass: 'menu-panel-container',
+ *         focusOnOpen: false,
+ *         zIndex: 100,
+ *         propagateContainerEvents: true,
+ *         groupName: 'menus'
+ *       }
+ *     };
+ *
+ *     $mdPanelConfigProvider.createConfig(config);
+ *   }
+ *
+ *   function DemoCtrl($mdPanel) {
+ *     $mdPanel.newPanelGroup('menus', {
+ *       maxOpen: 1
+ *     });
+ *
+ *     this.showMenu = function($event) {
+ *       var config = $mdPanel.getConfig('demoConfig');
+ *       config.id = 'menu_demo';
+ *       config.position = $mdPanel.newPanelPosition()
+ *           .relativeTo($event.srcElement)
+ *           .addPanelPosition(
+ *             $mdPanel.xPosition.ALIGN_START,
+ *             $mdPanel.yPosition.BELOW
+ *           );
+ *       config.locals = {
+ *         items: {
+ *           'Account',
+ *           'Sign Out'
+ *         }
+ *       };
+ *       config.openFrom = $event;
+ *
+ *       $mdPanel.open(config);
+ *     };
+ *   }
+ *
+ *   function DemoMenuCtrl(mdPanelRef) {
+ *     this.closeMenu = function() {
+ *       mdPanelRef && mdPanelRef.close();
+ *     };
+ *   }
+ * })(angular);
+ * </hsjs>
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanelConfig#createConfig
+ * @description
+ * Takes the passed in panel configuration object and adds it to the `_configs`
+ * object of the provider.
+ *
+ * @param {!Object<string, !Object>} config Specific configuration object that
```

Comment:
I'm not sure if we defined a type for the config, but if we did it would be useful to write here along with which options are not supported.